File: logipack-hub/hub-web/src/lib/server/hubApi/dto/audit.ts
export type AuditEventDto = {
	id: string;
	occurred_at: string;
	actor_user_id?: string | null;
	actor_display_name?: string | null;
	action_key: string;
	entity_type?: string | null;
	entity_id?: string | null;
	entity_label?: string | null;
	office_id?: string | null;
	office_label?: string | null;
	target_route?: string | null;
	metadata?: Record<string, unknown> | null;
};

export type AuditPageDto = {
	limit: number;
	next_cursor?: string | null;
	has_next: boolean;
};

export type ListAuditResponseDto = {
	events: AuditEventDto[];
	page: AuditPageDto;
};

File: logipack-hub/hub-api/src/dto/audit.rs
use serde::{Deserialize, Serialize};
use serde_json::Value as JsonValue;

#[derive(Debug, Serialize, Deserialize)]
pub struct AuditEventDto {
    pub id: String,
    pub occurred_at: String,
    pub actor_user_id: Option<String>,
    pub actor_display_name: Option<String>,
    pub action_key: String,
    pub entity_type: Option<String>,
    pub entity_id: Option<String>,
    pub entity_label: Option<String>,
    pub office_id: Option<String>,
    pub office_label: Option<String>,
    pub target_route: Option<String>,
    pub metadata: Option<JsonValue>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct AuditPageDto {
    pub limit: u64,
    pub next_cursor: Option<String>,
    pub has_next: bool,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ListAuditResponse {
    pub events: Vec<AuditEventDto>,
    pub page: AuditPageDto,
}
